docs/modules/ROOT/pages/how-tos/worklows/git-submodules.adoc

```diff
@@ -0,0 +1,128 @@
+ENV <SUBMODULEPKG> /path/to/submodule
+
+# Install necessary dependencies
+RUN dnf -y install <dependencies>
+
+# Set the working directory
+WORKDIR ${<SUBMODULEPKG>}
+
+RUN sed -i 's/<old-text>/<new-text>/g' /file/in/submodule/directory
+
+WORKDIR /app
+
+ENTRYPOINT ["<entrypoint-executable>"]
+----
+
+=== Create the git submodule for the upstream repo to track it
+
+After creating the downstream repo, add a git submodule for the upstream repo:
+
+[source, bash]
+----
+git submodule add <upstream-url>
+----
+
+Check and configure the git submodule config:
+[source, gitmodules]
+----
+# .gitmodules
+[submodule "<project-name>"]
+  path = <path-in-downstream-repo>
+  url = https://github.com/<namespace>/<project>.git
+----
+See the link:https://git-scm.com/docs/gitsubmodules[Git documentation] for more details on configuring git submodules.
+
+**Note**: By default the branch is not specified in the git submodule configuration file when adding a git submodule.
+see link:https://git-scm.com/docs/gitsubmodules[Git documentation] for more details on how to configure the branches for the submodules.
+
+=== Onboard the component onto {ProductName}
+
+After creating the downstream repository you can onboard the component. See xref:/how-tos/creating.adoc[Creating applications and components] for more details.
+
+=== Using private repositories as a git submodule
+
+==== Github
+
+When using private repositories we encourage that you give the link:https://github.com/apps/red-hat-konflux[Red Hat {ProductName} Github app] access to that private repository.
+
+==== Gitlab
+
+For Gitlab repositories please see xref:/how-tos/configuring/creating-secrets.adoc#creating-source-control-secrets[Creating secrets for your builds]
+
+== Benefits of this guideline
+
+1. **Separation of Concerns**: The downstream repo can stay focused on its specific development goals while the submodule handles its own code. This keeps your downstream repo clean and modular.
+2. **Automatic Dependency Syncing with link:https://github.com/renovatebot/renovate[Renovate]**: By automating the dependency sync process, whether its syncing the git submodule, the Containerfile, etc. your downstream repo can always use the latest code from the upstream repo(s), ensuring that your project benefits from new features, bug fixes, and security updates.
+3. **Version Control for upstream repos**: You can lock the submodule to a specific commit or version, giving you control over when and how updates are integrated. This is especially useful if an upstream update introduces breaking changes.
+4. **{ProductName} CI**: Automated building and testing of your software artifacts help ensure that changes in the upstream repo don't break your downstream project.
+
+== Potential drawbacks
+
+1. **Complexity**: Managing submodules and automating their updates adds complexity to your workflow, such as:
+  - Debugging issues related to submodule updates
+  - Your automation configuration can be more challenging.
```

Comment:
Do you want to call out the potential for drift between the upstream and downstream repositories?

Reply:
Potential for drift has been mentioned: https://github.com/konflux-ci/docs/pull/120/commits/f1be568ae08b9605e685c3d1e1e288862518855a#diff-46f8b5b0ae481a31c84644c1fde822df6be41954ac5d4002de9444fe1b91fd08R99

Reply:
I could add a link to your doc as well on this line if you think that's a good idea?

Reply:
Let's leave the link out for now. We haven't linked to that document yet from the konflux-ci.dev/docs site ... unless you feel like we should. We can also add the information to the docs here and link to it from my work.